### PR TITLE
Simplify diffusion test into single test file

### PR DESCRIPTION
**Pull request by alexanderianblair**
Previously the diffusion regression test being used in Platypus solved both an MFEM and MOOSE variation of the problem, and compared the outputs via the L2 error norm, which required a MultiApp simulation and a Transfer to be called. This obfuscates the steps of the user setup of the MFEM FE problem.

This PR updates this test to consist of a single test file that uses MOOSE's VTKDiff tester to compare the output .pvtu files against reference data. It also removes the version of the diffusion problem in `test/tests/outputs` that was an effective duplicate of the main regression test for usage of the diffusion integrator in `test/tests/kernels`.

A minor fix to the registered base identifier in MFEMVariable has also been applied, to ensure MFEM variables can be added as AuxVariables to the problem as before.

Additionally, an option to specify the VTK output format has been added to the ParaView DataCollection wrapper. This is to allow an ASCII format to be used for tests with small output file sizes, which facilitates easier diagnosis of issues upon test failure and appears more robust when running MOOSE's XMLDiff tests on different systems, at the expense of test runtime.

### CODE REVIEW

**Comment by alexanderianblair** [comment]
Note - code coverage is erroneously low for this PR as the CI is skipping the new diffusion test, since `xmltodict` and `deepdiff` need to be available for MOOSE's XMLDiff tester to work. These are added to the platypus-deps docker build in this same PR.

File: docker/platypus-deps/Dockerfile [hunk 1]
```diff
@@ -51,6 +51,8 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get install
 RUN pip install --no-cache-dir packaging && \
     pip install --no-cache-dir pyyaml && \
     pip install --no-cache-dir setuptools && \
+    pip install --no-cache-dir xmltodict && \
```
Comment (Heinrich-BR) [inline docker/platypus-deps/Dockerfile:54]: Perhaps this should be in a different PR, but since these new packages are being installed here, we'll probably have to add them to the ampere build script.

Comment (alexanderianblair) [reply docker/platypus-deps/Dockerfile:54]: Good spot - worth adding to the sapphire-rapids build too, if they're not currently available on CSD3

Comment (alexanderianblair) [reply docker/platypus-deps/Dockerfile:54]: Updated Ampere build script with required python dependencies for the XMLDiff test. Also switched to using an ASCII VTK output format here, to help diagnosis of issues on test failure, at the expense of test runtime. 

The ampere build has been tested successfully on CSD3.

**Review by Heinrich-BR — APPROVED** [review]
